greenheart/simulation/technologies/steel.py

```diff
@@ -0,0 +1,970 @@
+        labor_cost_fivemonth (float):
+            Cost of labor for the first five months of operation, often used in startup
+            cost calculations.
+        maintenance_materials_onemonth (float):
+            Cost of maintenance materials for one month of operation.
+        non_fuel_consumables_onemonth (float):
+            Cost of non-fuel consumables for one month of operation.
+        waste_disposal_onemonth (float):
+            Cost of waste disposal for one month of operation.
+        monthly_energy_cost (float):
+            Cost of energy (electricity, natural gas, etc.) for one month of operation.
+        spare_parts_cost (float):
+            Cost of spare parts as part of the initial investment.
+        misc_owners_costs (float):
+            Miscellaneous costs incurred by the owner, including but not limited to,
+            initial supply stock, safety equipment, and initial training programs.
+    """
+
+    total_plant_cost: float
+    total_fixed_operating_cost: float
+    labor_cost_fivemonth: float
+    maintenance_materials_onemonth: float
+    non_fuel_consumables_onemonth: float
+    waste_disposal_onemonth: float
+    monthly_energy_cost: float
+    spare_parts_cost: float
+    misc_owners_costs: float
+
+
+def run_steel_model(plant_capacity_mtpy: float, plant_capacity_factor: float) -> float:
+    """
+    Calculates the annual steel production in metric tons based on plant capacity and
+    capacity factor.
+
+    Args:
+        plant_capacity_mtpy (float):
+            The plant's annual capacity in metric tons per year.
+        plant_capacity_factor (float):
+            The capacity factor of the plant.
+
+    Returns:
+        float: The calculated annual steel production in metric tons per year.
+    """
+    steel_production_mtpy = plant_capacity_mtpy * plant_capacity_factor
+
+    return steel_production_mtpy
+
+
+def run_steel_cost_model(config: SteelCostModelConfig) -> SteelCostModelOutputs:
+    """
+    Calculates the capital expenditure (CapEx) and operating expenditure (OpEx) for
+    a steel manufacturing plant based on the provided configuration.
+
+    Args:
+        config (SteelCostModelConfig):
+            Configuration object containing all necessary parameters for the cost
+            model, including plant capacity, feedstock costs, and integration options
+            for oxygen and heat.
+
+    Returns:
+        SteelCostModelOutputs: An object containing detailed breakdowns of capital and
+        operating costs, as well as total plant cost and other financial metrics.
+
+    Note:
+        The calculation includes various cost components such as electric arc furnace
+        (EAF) casting, shaft furnace, oxygen supply, hydrogen preheating, cooling tower,
+        and more, adjusted based on the Chemical Engineering Plant Cost Index (CEPCI).
+    """
+    feedstocks = config.feedstocks
+
+    model_year_CEPCI = 596.2
```

Comment:
As in ammonia model, we should at least make an issue for setting these values/figureing out what year these are for

Reply:
Better to just ask Evan the year and move on